src/components/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { CustomIconText } from 'src/components/CustomIconText'
+import { getTxTo } from 'src/routes/safe/components/Transactions/TxList/utils'
+
+import styled from 'styled-components'
+import Spacer from 'src/components/Spacer'
+import { useSelector } from 'react-redux'
+import { GATEWAY_URL } from 'src/utils/constants'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { lastViewedSafe } from 'src/logic/currentSession/store/selectors'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import { useTransactionType } from 'src/routes/safe/components/Transactions/TxList/hooks/useTransactionType'
+import { useKnownAddress } from 'src/routes/safe/components/Transactions/TxList/hooks/useKnownAddress'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { getChainById } from 'src/config'
+import { useAssetInfo } from 'src/routes/safe/components/Transactions/TxList/hooks/useAssetInfo'
+import { TxInfo } from 'src/routes/safe/components/Transactions/TxList/TxCollapsed'
+
+const MAX_TXS_DISPLAY = 5
+
+const PendingTxsList = (): ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const defaultSafe = useSelector(lastViewedSafe) || undefined
+
+  const { isLoading: isLoadingOwnerSafes, ownerSafes } = useOwnerSafes()
+  const [loadingOwnerSafes, setLoadingOwnerSafes] = useState<boolean>()
+
+  // Selected Safe should come from the SideBar task when it's done
+  const [selectedSafe, setSelectedSafe] = useState<string>()
+
+  const [isFetchingPendingTxs, setIsFetchingPendingTxs] = useState<boolean>()
+  const [queuedTransactions, setQueuedTransactions] = useState<TransactionSummary[]>([])
+
+  useEffect(() => {
+    if (isLoadingOwnerSafes === undefined) return
+
+    if (ownerSafes[chainId]?.length) {
+      setSelectedSafe(ownerSafes[chainId][0])
+    } else {
+      setSelectedSafe(defaultSafe)
+    }
+    setLoadingOwnerSafes(false)
+  }, [chainId, defaultSafe, isLoadingOwnerSafes, ownerSafes, selectedSafe])
+
+  // Fetch txs awaiting confirmations
+  useEffect(() => {
+    if (selectedSafe === undefined) return
+
+    let isCurrent = true
+    const fetchQueuedTxs = async () => {
+      const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(selectedSafe))
+      const txs = results
+        .filter((result) => isTransactionSummary(result))
+        .slice(0, MAX_TXS_DISPLAY)
+        .map((item: TransactionListItem) => (item as Transaction).transaction)
+      if (isCurrent) {
+        setQueuedTransactions(txs)
+        setIsFetchingPendingTxs(false)
+      }
+    }
+    setIsFetchingPendingTxs(true)
+    fetchQueuedTxs()
+
+    return () => {
+      isCurrent = false
+    }
+  }, [chainId, selectedSafe])
+
+  if (isFetchingPendingTxs) {
+    return (
+      <>
+        {Array.from(Array(MAX_TXS_DISPLAY).keys()).map((_, idx) => (
+          <TransactionSkeleton key={idx} />
```

Comment:
Using indexs for keys is bad practice. It is very easy to have duplicate keys by doing so.

Reply:
I can use the elements in the array in this case.

Reply:
As long as it's not just a number, it's safer. You could even do something fancy, e.g.
```suggestion
        {Array.from(Array(MAX_TXS_DISPLAY).keys()).map((_, idx, arr) => (
          <TransactionSkeleton key={arr.join('_').idx} />
```